swarm_dial.go

```diff
@@ -158,7 +159,11 @@ func (s *Swarm) Dial(ctx context.Context, p peer.ID) (*Conn, error) {
 		return nil, ErrDialToSelf
 	}
 
-	return s.gatedDialAttempt(ctx, p)
+	conn, err := s.gatedDialAttempt(ctx, p)
+	if s.protec != nil && strings.Contains(err.Error(), "privnet") {
+		log.Warninf("error: %s. Possible missmatched private network key", err.Error())
+	}
+	return conn, err
```

Comment:
It's `log.Warningf()` -- I think this is also a good case for error types in go-libp2p-interface-pnet. There is a dependency here, we're just hiding it away in a string.